### ISSUE TITLE
hake-language-c-0.8.1 test suite fails on NixOS because of undeclared system dependencies

### ISSUE THREAD

**Issue by peti** [samplecount/shake-language-c#26]
Citing from http://hydra.cryp.to/build/1330346/nixlog/1/raw:

~~~~~
Test suite spectests: RUNNING...

Host toolchain
  compiles a C file to an executable FAILED [1]
  compiles a C++ file to an executable FAILED [2]

Failures:

  tests/spectests.hs:51:
  1) Host toolchain compiles a C file to an executable
       uncaught exception: ShakeException (Error when running Shake build system:
       * tests/build/host_toolchain_compile_c/output/result.txt
       * tests/build/host_toolchain_compile_c/output/result
       arch: readCreateProcess: runInteractiveProcess: exec: does not exist (No such file or directory)
       )

  tests/spectests.hs:71:
  2) Host toolchain compiles a C++ file to an executable
       uncaught exception: ShakeException (Error when running Shake build system:
       * tests/build/host_toolchain_compile_cpp/output/result.txt
       * tests/build/host_toolchain_compile_cpp/output/result
       arch: readCreateProcess: runInteractiveProcess: exec: does not exist (No such file or directory)
       )

Randomized with seed 123489781

Finished in 0.0144 seconds
2 examples, 2 failures
Test suite spectests: FAIL
~~~~~

**Comment by kaoskorobase** [comment]
Can I fix this? Or is it a NixOS bug?

**Comment by peti** [comment]
Well, the build assumes that a program called `arch` is in `$PATH`, but it is not. Personally, I don't know what `arch` is and none of my machines have such a program installed, so I suppose it's not necessarily a problem that only NixOS users will run into.

**Comment by kaoskorobase** [comment]
`arch` is part of [coreutils](https://www.gnu.org/software/coreutils/manual/html_node/arch-invocation.html#arch-invocation). This package is usually installed as a dependency of a C compiler, which is required by `shake-language-c` and to run the tests. Is there anything I can do about that on my end, e.g. by declaring a dependency in the Cabal file?

**Comment by peti** [comment]
> `arch` is part of `coreutils`.

Not on my Linux machine:

~~~~
$ ls /nix/store/8xfxnm4a7jvyh0i1sd967z6rws8lh21a-coreutils-8.24/bin
[         cp         env     install  mktemp   pr        sha1sum    stdbuf   true      wc
base64    csplit     expand  join     mv       printenv  sha224sum  stty     truncate  who
basename  cut        expr    kill     nice     printf    sha256sum  sum      tsort     whoami
cat       date       factor  link     nl       ptx       sha384sum  sync     tty       yes
chcon     dd         false   ln       nohup    pwd       sha512sum  tac      uname
chgrp     df         fmt     logname  nproc    readlink  shred      tail     unexpand
chmod     dir        fold    ls       numfmt   realpath  shuf       tee      uniq
chown     dircolors  groups  md5sum   od       rm        sleep      test     unlink
chroot    dirname    head    mkdir    paste    rmdir     sort       timeout  uptime
cksum     du         hostid  mkfifo   pathchk  runcon    split      touch    users
comm      echo       id      mknod    pinky    seq       stat       tr       vdir
~~~~

> Is there anything I can do about that on my end, e.g. by declaring a dependency in the Cabal file?

My guess is that calling `uname -m`  instead would do the trick. `uname` is a standard POSIX command, but `arch` is not.

**Comment by kaoskorobase** [comment]
Great, thanks, I'll switch to `uname` then!

**Comment by peti** [comment]
Will you release a new version to Hackage?

**Comment by kaoskorobase** [comment]
I've just released version 0.8.4 on Hackage. Thanks again for the bug report!
